extra-cpp-bindings/src/wallectconnectregistry.rs

```diff
@@ -0,0 +1,168 @@
+use serde::{Deserialize, Serialize};
+use std::collections::BTreeMap;
+
+use crate::error::GameSdkError;
+use crate::{ImageUrl, Platform, WalletEntry};
+
+#[derive(Serialize, Deserialize)]
+pub(crate) struct Registry {
+    listings: BTreeMap<String, Listing>,
+    count: i64,
+    total: i64,
+}
+
+impl Registry {
+    pub(crate) fn fetch_new() -> Result<Self, GameSdkError> {
+        const URL: &str = "https://registry.walletconnect.com/api/v2/wallets";
+        let client = reqwest::blocking::Client::new();
+        let resp: Registry = client.get(URL).send()?.json()?;
+        Ok(resp)
+    }
+}
+
+impl Default for Registry {
+    fn default() -> Self {
+        // get default cached value
+        serde_json::from_str(include_str!("registry.json")).unwrap()
```

Comment:
how about load instead of include_str? Seems load the json is better in case of any update on registry.json in the future.

Reply:
dynamic loading may fail if the file is missing or invalid... perhaps it can check if that, and use the `include_str` as a fallback?